### ISSUE TITLE
SSHConnection - Lookup of host-keys and storage of host keys doesn't support non-standard SSH ports

### ISSUE THREAD

**Issue by MitchDrage** [greenbone/python-gvm#924]
### Expected behavior
SSH host-keys should be stored and retreved as such for connections on non-standard ports:
[ssh.example.com]:2222 ssh-ed25519 ABCDEF`

### Current behavior
Connections to non-standard SSH ports are stored without the square brackets and trailing port number.
Similarly, if a host key is stored using the OpenSSH standard (including square brackets and port number), the SSHConnection treats the key as not matching.

Here is a quick python script to recreate the issue:
```
from paramiko.client import SSHClient, AutoAddPolicy
from gvm.connections import SSHConnection

# Using Paramiko to store the key for the example
client = SSHClient()
client.load_host_keys(os.path.expanduser('~/.ssh/known_hosts'))
client.set_missing_host_key_policy(AutoAddPolicy())
client.connect(hostname="ssh.example.com, port=2222, username="username, password="password)
client.close()

# The resulting known_hosts file looks like this after the above commands:
# [ssh.example.com]:2222ssh-ed25519 ABCDEF

# Now using GVM
gvm_connection = SSHConnection(hostname="ssh.example.com", port=2222, username="username", password="password", known_hosts_file=(os.path.expanduser('~/.ssh/known_hosts')))

with Gmp(connection=gvm_connection) as gmp:
    print(gmp.get_version())
```

The resulting known_hosts file looks like this:
```
[ssh.example.com]:2222ssh-ed25519 ABCDEF
ssh.example.com ssh-ed25519 ABCDEF
```
Where the first line is the key stored by OpenSSH, and the second line is that stored by python-gvm.

### Steps to reproduce
1. See above script.

### GVM versions
Latest pull of Greenbone Community Containers.
Python-GVM v22.9.1

### Environment
Ubuntu, with Greenbone Community Containers using (mostly) the docker-compose.yml from [here](https://greenbone.github.io/docs/latest/22.4/container/index.html#docker-compose-file)
Changes include installing OpenSSH on the GVMD container and exposing gvmd.sock over SSH.

**Installation method / source:** As above.

### Logfiles

See example Python script above and the resulting known_hosts file output above.

**Comment by bjoernricks** [comment]
Implemented with #925 and released with python-gvm 23.2.0

**Comment by bjoernricks** [comment]
Thanks for taking care and providing a PR!
